test/functional/feature_reindex_readonly.py

```diff
@@ -34,11 +33,13 @@ def reindex_readonly(self):
         filename = self.nodes[0].chain_path / "blocks" / "blk00000.dat"
         filename.chmod(stat.S_IREAD)
 
-        used_chattr = False
-        if platform.system() == "Linux":
+        undo_immutable = lambda: None
+        # Linux
+        try:
+            subprocess.run(['chattr'], stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
             try:
                 subprocess.run(['chattr', '+i', filename], capture_output=True, check=True)
```

Comment:
On Ubuntu 22.04.3 LTS, this line throws an exception for me
```
2023-10-25T19:35:49.783000Z TestFramework (DEBUG): Make the first block file read-only
2023-10-25T19:35:49.785000Z TestFramework (WARNING): Command '['chattr', '+i', PosixPath('/tmp/bitcoin_func_test_i3g4dkjv/node0/regtest/blocks/blk00000.dat')]' returned non-zero exit status 1.
2023-10-25T19:35:49.785000Z TestFramework (WARNING): stderr: b'chattr: Operation not permitted while setting flags on /tmp/bitcoin_func_test_i3g4dkjv/node0/regtest/blocks/blk00000.dat\n'
```
In spite of this, the file is still changed to readonly, so the test still works as intended (it just doesn't set `undo_immutable` in the next line because of the exception).